### ISSUE TITLE
Upgrade completely broken on Ubuntu. lxc-docker no longer valid package name.  

### ISSUE THREAD

**Issue by nathanleclaire** [docker/machine#1681]
Due to the new package archives this: https://github.com/docker/machine/blob/master/libmachine/provision/ubuntu.go#L68 is wrong (therefore upgrade is broken).

We need to change this code to reference `engine-docker` and figure out how to address the situation with current users.

See: https://blog.docker.com/2015/07/new-apt-and-yum-repos/ for how to migrate from `lxc-docker` to `docker-engine`.

cc @ehazlett

**Comment by tianon** [comment]
I actually see a few things in that function that worry me (like `apt-get upgrade some-package`, which doesn't actually upgrade just `some-package` and will really try to upgrade all packages on the system).

Here's what I'd propose:

`install` + `upgrade`:

1. always overwrite `/etc/sources.list.d/docker.list`
2. always `apt-get update`
3. always `apt-get install -y docker-engine` (which will both install or upgrade as appropriate)

`remove`:

1. same as today, although possibly slightly more like `apt-get remove -y docker-engine lxc-docker` to make sure we cover our bases

Installing `docker-engine` _should_ automatically remove `lxc-docker`, but that's worth some testing (ie, don't just take my word for it :innocent:).

**Comment by mahnunchik** [comment]
+1

```bash
$ docker-machine upgrade test-machine
SSH cmd error!
command: DEBIAN_FRONTEND=noninteractive sudo -E apt-get upgrade -y  lxc-docker
err    : exit status 100
output : Reading package lists...
Building dependency tree...
Reading state information...
Package lxc-docker is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'lxc-docker' has no installation candidate
```

I've tried to upgrade machine from 1.8.0 to 1.8.1

**Comment by nathanleclaire** [comment]
Sorry for the trouble @mahnunchik.  Patch submitted by the incorrigible @ehazlett here: https://github.com/docker/machine/pull/1687

**Comment by ehazlett** [comment]
@tianon thx!  I don't know how i missed the `upgrade` thing :(  That's fixed in #1687.  I didn't implement the dual package logic in `remove` as we do not ever "remove" Docker and I want to try to keep the backward logic out.  If a user is on an old version of the install, they simply will just live with it.  We do not offer a way to remove it to hit that case.  If they upgrade, the upgrade will remove it for them and then it should be good.

However, if there is strong opposition to this, I will add in the remove logic.

Thanks!

**Comment by ehazlett** [comment]
Closing as #1687 is merged.
